source/common/quic/platform/quiche_flags_impl.h

```diff
@@ -86,13 +94,29 @@ template <typename T> class TypedFlag : public Flag {
   // Return flag value.
   T value() const {
     absl::MutexLock lock(&mutex_);
+    if (has_reloaded_value_) {
+      return reloaded_value_;
+    }
     return value_;
   }
 
+  void setReloadedValue(T value) {
+    absl::MutexLock lock(&mutex_);
+    has_reloaded_value_ = true;
+    reloaded_value_ = value;
+  }
+
+  void resetReloadedValue() override {
+    absl::MutexLock lock(&mutex_);
+    has_reloaded_value_ = false;
+  }
+
 private:
   mutable absl::Mutex mutex_;
   T value_ ABSL_GUARDED_BY(mutex_);
   T default_value_;
```

Comment:
pre-existing nit: I think that default_value can be marked const.
`const T default_value_;`